flint/flint-spark-integration/src/main/scala/org/apache/spark/sql/flint/FlintDataSourceV2.scala

```diff
@@ -3,35 +3,42 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-package org.apache.spark.sql.v2
+package org.apache.spark.sql.flint
 
 import java.util
 
+import org.opensearch.flint.core.FlintOptions
+
 import org.apache.spark.sql.SparkSession
 import org.apache.spark.sql.connector.catalog.{Table, TableProvider}
 import org.apache.spark.sql.connector.expressions.Transform
 import org.apache.spark.sql.sources.DataSourceRegister
 import org.apache.spark.sql.types.StructType
 import org.apache.spark.sql.util.CaseInsensitiveStringMap
 
-class OpenSearchJsonDataSourceV2 extends TableProvider with DataSourceRegister {
+class FlintDataSourceV2 extends TableProvider with DataSourceRegister {
+
+  private var table: FlintTable = null
 
-  /**
-   * Todo. add inferSchema for read path
-   */
   override def inferSchema(options: CaseInsensitiveStringMap): StructType = {
-    throw new UnsupportedOperationException("inferSchema is not supported")
+    if (table == null) {
+      table = getFlintTable(Option.empty, new FlintOptions(options.asCaseSensitiveMap()))
+    }
```

Comment:
because we "cache" table, I assume the caller on `inferSchema` and `getTable` will always pass same arguments?

Reply:
done.
follow the [FileDataSourceV2](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileDataSourceV2.scala#L105).